### PR TITLE
Handle undefined message when sending AT command

### PR DESCRIPTION
**Pull request by chinglee-iot**
This PR fix the issue feedback in [FreeRTOS forum](https://forums.freertos.org/t/cellular-interface-receiving-unexpected-unhandled-response-after-a-command-response-makes-next-command-fail/15889/2).

* Update pktio to handle undefined message when sending AT command
* Undefined message handler is moved to pkthandler
* Use PktRespMutex in pktio thread to keep data consistency
* Add test cases to cover the undefined message handle flow
* Update ATE0 handling in SARA R4 and HL7802. ATE0 may still echo back ATE0. Change the AT command type to CELLULAR_AT_WO_PREFIX

### CODE REVIEW

File: source/cellular_pkthandler.c [hunk 1]
```diff
@@ -93,7 +93,6 @@ static CellularPktStatus_t _convertAndQueueRespPacket( CellularContext_t * pCont
     if( ( pBuf != NULL ) )
     {
         pAtResp = ( const CellularATCommandResponse_t * ) pBuf;
-        PlatformMutex_Lock( &pContext->PktRespMutex );
```
Comment (ActoryOu) [inline source/cellular_pkthandler.c:96]: Why do we need to remove the lock here?
We still need to read pktRespCB in pContext. Don't we need to protect this part?

Comment (chinglee-iot) [reply source/cellular_pkthandler.c:96]: Yes, the packet handler callback is called by pktio thread. Pktio thread acquires the lock already.

File: source/cellular_pktio.c [hunk 2]
```diff
@@ -1046,7 +1064,11 @@ static void _pktioReadThread( void * pUserData )
                 /* Keep Reading until there is no more bytes in comm interface. */
                 do
                 {
+                    /* When pktio thread is parsing the response from modem, pktio
+                     * thread holds the PktRespMutex until no RX data from comm interface. */
+                    PlatformMutex_Lock( &pContext->PktRespMutex );
                     bytesRead = _handleRxDataEvent( pContext, &pAtResp );
```
Comment (ActoryOu) [inline source/cellular_pktio.c:1070]: Lock whole rx event here. 
But we need to call callback functions to users and they're not handled by ourselves.
Is it possible to decouple them?

**Review by ActoryOu — CHANGES_REQUESTED** [review]

**Review by ActoryOu — DISMISSED** [review]

File: modules/hl7802/cellular_hl7802.c [hunk 3]
```diff
@@ -335,6 +335,15 @@ CellularError_t Cellular_ModuleEnableUE( CellularContext_t * pContext )
...
+    CellularAtReq_t atReqGetWoPrefix =
+    {
+        NULL,
+        CELLULAR_AT_WO_PREFIX,
+        NULL,
+        NULL,
+        NULL,
+        0
+    };
     CellularAtReq_t atReqGetNoResult =
```
Comment (ActoryOu) [inline modules/hl7802/cellular_hl7802.c:347]: Do we still need this variable in this function?

File: modules/sara_r4/cellular_r4.c [hunk 4]
```diff
@@ -262,7 +262,15 @@ CellularError_t rebootCellularModem( CellularContext_t * pContext,
...
+    CellularAtReq_t atReqGetWoPrefix =
+    {
+        NULL,
+        CELLULAR_AT_WO_PREFIX,
+        NULL,
+        NULL,
+        NULL,
+        0
+    };
     CellularAtReq_t atReqGetNoResult =
```
Comment (ActoryOu) [inline modules/sara_r4/cellular_r4.c:274]: Do we still need this variable in this function?

File: modules/hl7802/cellular_hl7802.c (hunk 3 above)
Comment (chinglee-iot) [reply modules/hl7802/cellular_hl7802.c:347]: When modem in echo mode, ATE0 command will be echoed back. This command should be regarded as CELLULAR_AT_WO_PREFIX. Other command will still be regarded as CELLULAR_AT_NO_RESULT command.
```
> ATE0
< ATE0
< OK
```

File: modules/sara_r4/cellular_r4.c (hunk 4 above)
Comment (chinglee-iot) [reply modules/sara_r4/cellular_r4.c:274]: When modem in echo mode, ATE0 command will be echoed back. This command should be regarded as CELLULAR_AT_WO_PREFIX. Other command will still be regarded as CELLULAR_AT_NO_RESULT command.
```
> ATE0
< ATE0
< OK
```

**Review by ActoryOu — APPROVED** [review]

**Comment by ActoryOu** [comment]
Please help to move modules change to below links:
https://github.com/FreeRTOS/Lab-FreeRTOS-Cellular-Interface-Reference-Sierra-Wireless-HL7802
https://github.com/FreeRTOS/Lab-FreeRTOS-Cellular-Interface-Reference-ublox-SARA-R4

Thanks.

**Review by shubnil — APPROVED** [review]

**Pull request merged**